tests/v2/validation/provisioning/rke2/etcd_backup_restore_test.go

```diff
@@ -420,12 +427,37 @@ func (r *RKE2EtcdSnapshotRestoreTestSuite) watchAndWaitForPods(client *rancher.C
 			}
 
 			phase := podStatus.Phase
-			if phase != v1.PodRunning && phase != v1.PodSucceeded {
-				return false, nil
+			conditions := podStatus.Conditions
+			if phase == v1.PodPending {
+				foundPendingPod = true
+			} else if phase == v1.PodFailed || phase == v1.PodUnknown {
+				podErrors = append(podErrors, fmt.Errorf("ERROR: %s: %s", pod.Name, podStatus))
+			} else if phase == v1.PodRunning {
+				podReady := false
+				for _, condition := range conditions {
+					if condition.Type == v1.PodReady && condition.Status == v1.ConditionTrue {
+						podReady = true
+						break
+					}
+				}
+				if !podReady {
```

Comment:
I understand this is specifically checking against the reported error with the `helm-operation` pods having a state of `CrashLookBackOff`. However, I think we should generalize the check to just check for an error?

@igomez06 what do you think of this?

Reply:
I've made the required change, now it keeps on polling if there's any error, below is the output from jenkins:
```
level=info msg="container helm of pod helm-install-rke2-snapshot-controller-6zwr8 is not ready, state: {Waiting:&ContainerStateWaiting{Reason:CrashLoopBackOff,Message:back-off 2m40s restarting failed container=helm pod=helm-install-rke2-snapshot-controller-6zwr8_kube-system(a0da95f6-41f4-43b5-8ab6-ba66c5f6b785),} Running:nil Terminated:nil} "
```

If it founds the pod ready it continues on checking the next pod.

Reply:
Ah I see this make sense, you want to check every pod. Is this the functionality you want? This looks good to me.

Reply:
Yeah that looks good, but a follow-up question from looking at the Jenkins log...if we were to have a timeout of say 60m, that message would keep looping until then, right?

The Jenkins output stops due to the default timeout being met. The change itself looks good, just want to make sure we aren't flooding Jenkins output with the same message.

Reply:
Yes that's correct @igomez06.
@markusewalker, the timeout is fixed in the polling [here](https://github.com/rancher/rancher/pull/41693/files#diff-81f7cd6e882c53e15dc01797231a49c4e4a8a911f1c1d0e22ae9576f5952fe38R399), which is 5 mins. If all the pods are ready before that, it will not wait and exit out from the poll